src/main/java/org/osc/manager/ism/api/IsmSecurityGroupApi.java

```diff
@@ -94,12 +96,12 @@ public void updateSecurityGroup(String mgrSecurityGroupId, String name, Security
 
     @Override
     public void deleteSecurityGroup(String mgrSecurityGroupId) throws Exception {
-        SecurityGroupEntity existingSg = (SecurityGroupEntity) getSecurityGroupById(mgrSecurityGroupId);
-        if (existingSg == null) {
-            throw new Exception(String.format("A security group with id %s was not found.", mgrSecurityGroupId));
-        }
-
         this.txControl.required(() -> {
+            SecurityGroupEntity existingSg = (SecurityGroupEntity) getSecurityGroupById(mgrSecurityGroupId);
```

Comment:
we dont want to throw an exception within a transaction since that gets wrapped with scoped exception.
see https://github.com/opensecuritycontroller/sdn-controller-nsfc-plugin/pull/11#discussion_r143563422

Please revert back the change.

To drive consistency between SG and SGI api we could log a warning and not throw an exception, like we do in

`org.osc.manager.ism.api.IsmSecurityGroupInterfaceApi.deleteSecurityGroupInterface(String)`

Reply:
@arvindn05 , this is changed as deletion was not working. I will investigate further why deletion does not work when transaction is just put around the remove part.

Reply:
@arvindn05 , I will implement the above changes once you approve the above solution.
Note: The above exception of detached instance was getting masked by the rest layer exception "IllegalStateException", so the above exception was never seen.

Reply:
sounds good. I believe that was the solution we came up with for the https://github.com/opensecuritycontroller/sdn-controller-nsfc-plugin/blob/master/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java#L292 as well.